File: 2271-rearrange-array-elements-by-sign/rearrange-array-elements-by-sign.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
        int i = 0 ; 
        int j = 1 ;
        vector<int> ans(nums.size());

        for(int k = 0 ; k < nums.size() ; k++){
            int temp = nums[k] ;
            if(temp  > 0){
                ans[i] = temp;
                i+=2;
            }else{
                ans[j] =  temp;
                j+=2;
            }
        }
        return ans;
    }
};

File: 2271-rearrange-array-elements-by-sign/rearrange-array-elements-by-sign.java
class Solution {
    public int[] rearrangeArray(int[] nums) {
        int n = nums.length;

        int[] pos = new int[n/2];
        int[] neg = new int[n/2];
        int posIdx = 0 ;
        int negIdx = 0 ;

        for(int i : nums){
            if(i < 0){
                neg[negIdx++] = i;
            }else{
                pos[posIdx++] = i;
            }
        }

        int j = 0 ;
        posIdx = 0 ;
        negIdx = 0 ;


        while(j < n){

            nums[j] = pos[posIdx];
            j++;
            nums[j] = neg[negIdx];
            j++;
            posIdx++;
            negIdx++;
        }

        return nums; 

    }
}